R/estimateDominance.R

```diff
@@ -0,0 +1,366 @@
+#'
+#' @param name A name for the column of the colData where the calculated
+#' Dominance indices should be stored in.
+#'
+#' @param BPPARAM A
+#'   \code{\link[BiocParallel:BiocParallelParam-class]{BiocParallelParam}}
+#'   object specifying whether calculation of estimates should be parallelized.
+#'
+#' @details
+#' \code{estimateDominance} calculates the community dominance indices.
+#'
+#' @return Returns ME object, that includes dominance indices in coldata.
+#'
+#' @seealso
+#' \itemize{
+#'   \item{\code{\link[=estimateDiversity]{estimateDiversity}}}
+#' }
+#'
+#' @name estimateDominance
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' data(esophagus, package = "MicrobiomeExperiment")
+#' esophagus <- as(esophagus, "MicrobiomeExperiment")
+#'
+#' #Calculates simpson dominance index
+#' esophagus <- estimateDominance(esophagus, index="simpson")
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' #Tries to calculate index that is not accepted. Gets an error.
+#' esophagus <- estimateDominance(esophagus, index="does_not_exist")
+#' #Tries to calculate index that is not accepted
+#' # and index that is accepted. Gets an error.
+#' esophagus <- estimateDominance(esophagus, index=c("dmn", "does_not_exist"))
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' #Calculates DBP and Core Abundance indices
+#' esophagus <- estimateDominance(esophagus, index=c("dbp", "core_abundance"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #Shows DBP index
+#' colData(esophagus)$dbp
+#'
+#' #Deletes DBP index
+#' colData(esophagus)$dbp <- NULL
+#' #Shows all indices, DBP is deleted
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Names of columns can be chosen, but the length of arguments must match.
+#' esophagus <- estimateDominance(esophagus,
+#'     index=c("dbp", "core_abundance"),
+#'     name = c("index1", "index2"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #If they do not match, gets an error.
+#' esophagus <- estimateDominance(esophagus,
+#'     index="simpson",
+#'     name = c("index3", "index4"))
+#' #Shows all indices
+#' colData(esophagus)
+#' #Deletes all indices
+#' colData(esophagus) <- NULL
+#'
+#' #Calculates all indices
+#' esophagus <- estimateDominance(esophagus)
+#' #Or you can calculate all indices by specify "all"
+#' esophagus <- estimateDominance(esophagus, index="all")
+#' #Shows all indices
+#' colData(esophagus)
+#'
+#' @export
+#'
+setGeneric("estimateDominance",signature = c("x"),
+           function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
+                    name = index, ...)
+               standardGeneric("estimateDominance"))
+
+
+#' @rdname estimateDominance
+#' @export
+setMethod("estimateDominance", signature = c(x = "MicrobiomeExperiment"),
+          function(x, abund_values = "counts", index="all", rank=1, as_relative=TRUE, aggregate=TRUE,
```

Comment:
Added testing. taxonomyRanks(x)[1L] to default value of rank

Reply:
Changed rank back to numerical values 1 or 2, because it is not a phylogenetic rank. It is used when calculating DBP and DMN. It stores the value 1 or 2 which represents if the most abundant taxon is taken into account (1) or if two most abundant taxa is taken into account (2).

Because rank is used represent phylogenetic rank in many other functions, it might be good idea to consider other name for that to avoid misunderstanding. 

Now, I changed the name of 'rank' to 'ntaxa'.